### PR TITLE
Add a section about implementation review

### PR DESCRIPTION
**Pull request by rjmccall**
This incorporates feedback received in the Language Workgroup meeting, as well as some of the ideas from https://github.com/apple/swift-evolution/pull/957.

### CODE REVIEW

File: review_management.md [hunk 1]
```diff
@@ -37,6 +37,38 @@ Cursory review is meant to be a rapid and lightweight "triage" process which kee
...
+
+- The implementation must fully implement the proposed feature. If part of the feature is proving harder to implement, that may indicate significant theoretical or practical problems; an incomplete implementation should not be ignored as something that can be finished later.
+
+- The implementation must take a sound conceptual approach as judged by code maintainers in the affected areas. It may have significant implementation flaws, such as not fitting into the compiler architecture correctly, but the basic idea of the implementation should not be dodging major theoretical concerns. Cursory reviewers should engage with appropriate code maintainers in order to get their opinion about whether the implementation satisfies this point.
+
+- The implementation needs to work well enough for its consequences to be tested, both by Swift users and by automated systems. All of the required steps here are achievable automatically with the [Swift CI system](https://www.swift.org/continuous-integration/).
+
+  - At a minimum, a PR enabling the feature must pass the full Swift project correctness test suite.
```
Comment (beccadax) [inline review_management.md]: By “the full Swift project correctness test suite”, do you mean the tests run by “swift-ci please test", or do you mean something else?

**Review by DougGregor — APPROVED** [review]
Modulo Becca's comment, this LGTM

**Review by Jumhyn — APPROVED** [review]
Content pretty much looks good to me.

Rhetorically, there's a couple things that I think we could make clearer. 

The section is titled "Cursory implementation review" but doesn't really directly address the cursory implementation _review_ at all. I think it would be good to add a sentence after "The initial proposal implementation [...] the three goals above," something like
> To that end, workgroup members should use the following guidelines when evaluating a proposal's prototype implementation to determine whether the proposal is ready to be scheduled for review:

Additionally, regarding "the three goals above" (also referred to "purposes" before they are listed), I think it would be good if we were very clear about the _specific goal_ at the beginning of each paragraph—right now you kind of have to read the whole paragraph to glean the underlying purpose, and I think this section would be clearer if it read as something like (new language in brackets):

> First, [to ensure that the proposal is theoretically sound]. Some proposals that seem reasonable...
>
> Second, [to ensure that proposals are able to be practically integrated with the existing language]. Some proposals that seem reasonable...
>
> Finally, [to ensure that proposals can be fully implemented promptly after their acceptance]. Evolution proposals are expected to build...

File: review_management.md (hunk 1 above)
Comment (rjmccall) [reply review_management.md]: Yes, exactly.  Clarified.

**Comment by rjmccall** [comment]
> Rhetorically, there's a couple things that I think we could make clearer.

Thanks, incorporated.

**Pull request merged**